server/base/src/main/java/org/apache/accumulo/server/manager/LiveTServerSet.java

```diff
@@ -227,16 +229,22 @@ public synchronized void scanServers() {
     try {
       final Set<TServerInstance> updates = new HashSet<>();
       final Set<TServerInstance> doomed = new HashSet<>();
+      final ZooCache zc = getZooCache();
+      final String tserverRoot = context.getZooKeeperRoot() + Constants.ZTSERVERS;
 
-      final String path = context.getZooKeeperRoot() + Constants.ZTSERVERS;
+      Set<ServiceLockPath> tservers = new HashSet<>();
 
-      HashSet<String> all = new HashSet<>(current.keySet());
-      all.addAll(getZooCache().getChildren(path));
+      for (String resourceGroup : zc.getChildren(tserverRoot)) {
```

Comment:
I made another comment suggesting that maybe all the methods to get servers should have a third argument specifying if the path must have children lock nodes or not.

Reply:
Yeah, I can make that change. That will likely help to remove the residual places in the code where the constants are being used and zookeeper paths being accessed outside of ServerZooKeeperPaths. I'll plan on doing that for #4883 so that I don't keep adding changes to this large PR.